src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,292 @@
+package scala
+package collection.mutable
```

Comment:
This repo, as far as I understand, is a sandbox/playground type repo to prototype ideas before it lands in Scala. I just need access to `.sizeHintIfCheap` - which would presumably exist in future Traversable too right? 

Making it use `collection.strawman` is also quite trivial. Just revert this commit: https://github.com/scala/collection-strawman/pull/49/commits/b955281ce2134ffd101a2dd920e16d440324f290

It does not really use any other Scala collection besides Arrays underneath...